File: src/main/java/com/ultreon/bubbles/common/random/EntityRandomizer.java
package com.ultreon.bubbles.common.random;

import com.ultreon.bubbles.common.EntityProperties;
import com.ultreon.bubbles.common.gametype.AbstractGameType;

import java.awt.geom.Rectangle2D;

/**
 * Abstract Entity Randomizer.
 */
public abstract class EntityRandomizer {
    public abstract EntityProperties getRandomProperties(Rectangle2D bounds, AbstractGameType gameType);
}

File: src/main/kotlin/qtech/bubbles/common/random/EntityRandomizer.kt
package qtech.bubbles.common.random

import qtech.bubbles.common.EntityProperties
import qtech.bubbles.common.gametype.AbstractGameMode
import java.awt.geom.Rectangle2D

/**
 * Abstract Entity Randomizer.
 */
abstract class EntityRandomizer {
    abstract fun getRandomProperties(bounds: Rectangle2D, gameMode: AbstractGameMode): EntityProperties
}